File: app/src/main/java/jp/ttanaka330/androidlearning/ui/fragment/BaseFragment.java
package jp.ttanaka330.androidlearning.ui.fragment;

import dagger.android.support.DaggerFragment;

/**
 * {@link android.support.v4.app.Fragment} を実装する際の基盤となるクラスです。
 * 本アプリでは DI を行うために {@link DaggerFragment} を継承しています。
 *
 * @see <a href="https://google.github.io/dagger/">Dagger2</a>
 */
public abstract class BaseFragment extends DaggerFragment {

    @Override
    public String toString() {
        return getClass().getSimpleName();
    }
}

File: learning/library/dagger/src/main/java/com/github/ttanaka330/learning/todo/BaseFragment.kt
package com.github.ttanaka330.learning.todo

import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import dagger.android.support.DaggerFragment

open class BaseFragment : DaggerFragment() {

    fun setToolBar(@StringRes titleId: Int, isHomeUpEnabled: Boolean = false) {
        (activity as? AppCompatActivity)?.supportActionBar?.let {
            it.setTitle(titleId)
            it.setDisplayHomeAsUpEnabled(isHomeUpEnabled)
        }
    }
}